eval/src/vespa/eval/eval/cell_type.h

```diff
@@ -25,14 +25,68 @@ template <typename CT> inline CellType get_cell_type();
 template <> inline CellType get_cell_type<double>() { return CellType::DOUBLE; }
 template <> inline CellType get_cell_type<float>() { return CellType::FLOAT; }
 
+struct CellTypeUtils {
+    template <typename CT> static constexpr bool check(CellType type);
+    template <typename CT> static constexpr CellType get_type();
+
+    static constexpr uint32_t DOUBLE_ID = 0;
```

Comment:
these should be owned by the value codec. (including id_of and type_of functions)